### PR TITLE
use Sled to store rbt database

### PR DESCRIPTION
**Pull request by BrianHicks**
Moves our data storage to use Sled, which will allow us to read/write in parallel pretty soon here.

Based on https://github.com/roc-lang/rbt/pull/77. Review/merge that first!

Fixes https://github.com/roc-lang/rbt/issues/72

### CODE REVIEW

**Comment by BrianHicks** [comment]
@roc-lang/rbt-reviewers does this look OK? Is sled too much here or are we OK to continue building on this?

File: src/coordinator.rs [hunk 1]
```diff
@@ -48,20 +47,7 @@ impl<'roc> Builder<'roc> {
...
-        let mut meta_to_hash: HashMap<u64, String> = match File::open(&file_hashes_path) {
-            Ok(file) => {
-                let reader = BufReader::new(file);
-                serde_json::from_reader(reader)
-                    .context("could not deserialize mapping from inputs to content")?
-            }
-            Err(err) if err.kind() == std::io::ErrorKind::NotFound => HashMap::default(),
-            Err(err) => return Err(err).context("could not open file hash cache"),
-        };
+        // let mut meta_to_hash: HashMap<u64, String> ;
```
Comment (bhansconnect) [inline src/coordinator.rs:50]: nit: remove line?

File: src/store.rs [hunk 2]
```diff
@@ -286,9 +268,9 @@ impl Item {
         }
     }
 
-    fn from_hex(root: &Path, hex: &str) -> Result<Self> {
-        let hash = blake3::Hash::from_hex(hex)
-            .with_context(|| format!("could not load a blake3 hash from hex value `{}`", hex))?;
+    fn from_hex(root: &Path, hex: impl AsRef<[u8]>) -> Result<Self> {
+        let hash =
+            blake3::Hash::from_hex(hex).context("could not load a blak3 hash from hex value")?;
```
Comment (bhansconnect) [inline src/store.rs:273]: nit: typo blak3 -> blake3

**Review by bhansconnect — APPROVED** [review]
Sorry for the slow review.

This seem pretty straight forward and reasonable.

**Comment by BrianHicks** [comment]
no worries, and thank you for the review!

**Pull request merged**